### PR TITLE
Add new warnoptions loadPyodide parameter

### PR DESCRIPTION
**Pull request by hoodmane**
Use it in the test suite to make `DeprecationWarning` into an error.

### CODE REVIEW

File: pyodide-test-runner/pyodide_test_runner/browser.py [hunk 1]
```diff
@@ -164,7 +164,7 @@ def javascript_setup(self):
     def load_pyodide(self):
         self.run_js(
             """
-            let pyodide = await loadPyodide({ fullStdLib: false, jsglobals : self });
+            let pyodide = await loadPyodide({ fullStdLib: false, jsglobals : self, warnoptions : "error::DeprecationWarning" });
             self.pyodide = pyodide;
             globalThis.pyodide = pyodide;
             pyodide._api.inTestHoist = true; // improve some error messages for tests
```
Comment (ryanking13) [inline pyodide-test-runner/pyodide_test_runner/browser.py:170]: For example, how about adding warnoptions here such as:

```
pyodide._module['PYTHONWARNINGS'] = ...
```

**Review by ryanking13 — COMMENTED** [review]
Thanks, @hoodmane! I think this is mostly for tests and not for Pyodide end users, so what do you think of adding it inside our test API instead of `loadPyodide`?

**Comment by hoodmane** [comment]
What if other people want to adjust warnings in their tests?

**Comment by ryanking13** [comment]
> What if other people want to adjust warnings in their tests?

I think people can use `warnings.filterwarnings` or set the PYTHONWARNINGS variable during runtime unless it is required to adjust warning filters during Pyodide load time.

WDYT @rth?

**Comment by hoodmane** [comment]
The `PYTHONWARNINGS` variable is only used at Python initialization, subsequent updates to it will be ignored. People could use `warnings.filterwarnings`. You could also use that in native Python but it provides a `-W` flag to pass warning settings anyways. Of the flags in `python --help` this one looks the most useful to me.

**Comment by hoodmane** [comment]
@ryanking13 ideally it would be good to set these warnings to error on our core tests and not on the package tests. You have any idea what the best way to handle this would be?

**Comment by ryanking13** [comment]
> You could also use that in native Python but it provides a `-W` flag to pass warning settings anyways. Of the flags in `python --help` this one looks the most useful to me.

Okay, I was a bit reluctant to add too many options in `loadPyodide`, but it seems like it is quite reasonable to have `warnoptions`, corresponding to `-W` flags in Python.

> ideally it would be good to set these warnings to error on our core tests and not on the package tests.

Yes, I'm +1 for it.

**Comment by rth** [comment]
So there is a [number env variables](https://docs.python.org/3/using/cmdline.html#environment-variables) that control Python behavior (with warnings probably being the most useful). I certainly see the use case for this, but what bothers me is that a) we can't keep up adding new parameters for other env variables, b) that the chosen name is non standard.

Maybe we could pass the `env` and set it to `env: {PYTHONWARNINGS`: ...}`?  It's a bit more verbose but more generic. I suppose these env variables need to be passed at init, right (and setting them later has no effect)?

**Comment by hoodmane** [comment]
> I suppose these env variables need to be passed at init, right (and setting them later has no effect)?

That is correct.

**Pull request closed without merging**